src/AzureClient/AzureExecutionTarget.cs

```diff
@@ -76,21 +76,58 @@ internal protected AzureExecutionTarget(string? targetId)
     ///     level.
     /// </summary>
     public bool SupportsCapability(TargetCapability capability) =>
-        TargetCapabilityModule.Subsumes(this.MaximumCapability, capability);
+        // NB: Duplicates logic at https://github.com/microsoft/qsharp-compiler/blob/7714168fda4379fb7e6a6c616f680ec039c482f4/src/QuantumSdk/DefaultItems/DefaultItems.targets#L78,
+        //     but at the level of providers rather than at the level of resolved processors.
+        (GetProvider(TargetId) switch
+        {
+            AzureProvider.Quantinuum or AzureProvider.Honeywell => new[]
+            {
+                TargetCapabilityModule.AdaptiveExecution,
+                TargetCapabilityModule.BasicMeasurementFeedback,
+                TargetCapabilityModule.BasicExecution
+            },
+            AzureProvider.IonQ => new[]
+            {
+                TargetCapabilityModule.BasicQuantumFunctionality
+            },
+            AzureProvider.QCI => new[]
+            {
+                TargetCapabilityModule.AdaptiveExecution,
+                TargetCapabilityModule.BasicExecution
+            },
+            AzureProvider.Rigetti => new[]
+            {
+                TargetCapabilityModule.BasicExecution
+            },
+            AzureProvider.Microsoft => new[]
+            {
+                TargetCapabilityModule.FullComputation
+            },
+            _ => new[]
+            {
+                TargetCapabilityModule.FullComputation
```

Comment:
All target capabilities should be listed here, i.e. add AdaptiveExecution, BasicExecution, BasicMeasurementFeedback and BasicQuantumFunctionality.

Reply:
These are checked against `Subsumes`, such that those four levels should also work here. Will add explicitly in any case.